src/amcl-extensions/big_256_56.c

```diff
@@ -84,12 +87,15 @@ void big_256_56_mod_mul_and_add(BIG_256_56 *big_out,
 }
 
 void big_256_56_random_mod_order(BIG_256_56 *big_out,
-                                 csprng *rng)
+                                 struct ecdaa_prng *prng)
 {
+    if (ECDAA_PRNG_INITIALIZED_YES != prng->initialized)
```

Comment:
Can all of these checks be moved into one function - an accessor for `ecdaa_prng->impl`?  Then we don't have to remember to check the initialized state in so many places; just the once in the accessor.

Reply:
And actually, in this particular case, because it is a `amcl-extension` it should take the `csprng` directly, not a `ecdaa_prng`. Ideally `acml-extensions` shouldn't depend on `ecdaa` code.

Reply:
That's a good idea.
:+1: